numpy/core/src/private/mem_overlap.c

```diff
@@ -535,15 +536,24 @@ solve_diophantine(unsigned int n, diophantine_term_t *E, npy_int64 b,
         return MEM_OVERLAP_NO;
     }
     else {
-        diophantine_term_t Ep[n];
-        npy_int64 Epsilon[n], Gamma[n];
+
+        diophantine_term_t *Ep = malloc(n * sizeof(diophantine_term_t));
+        npy_int64 *Epsilon = malloc(n * sizeof(npy_int64));
+        npy_int64 *Gamma = malloc(n *sizeof(npy_int64));
```

Comment:
could use npy_alloc_cache from alloc.h to avoid some overhead
the value also looks like its bounded by NPY_MAXDIMS, though not using might make removing it in future a tiny bit easier